server/conn.go

```diff
@@ -766,6 +826,14 @@ func (cc *clientConn) Run(ctx context.Context) {
 		cc.pkt.setReadTimeout(time.Duration(waitTimeout) * time.Second)
 		start := time.Now()
 		data, err := cc.readPacket()
+		if initConnectFailed {
```

Comment:
I am trying to match the semantic of when MySQL disconnects the client - but actually I just checked and it's not quite correct. See TiDB:

```
mysql> select 1;
ERROR 2013 (HY000): Lost connection to MySQL server during query
```

vs MySQL:

```
mysql [localhost:8023] {u1} ((none)) > select 1;
ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    48
Current database: *** NONE ***

ERROR 1184 (08S01): Aborted connection 48 to db: 'unconnected' user: 'u1' host: 'localhost' (init_connect command failed)
```

I might have misunderstood it, and it might occur in the initial handshake.

Reply:
I have changed it to the handshake stage. It is not quite identical to MySQL because MySQL seems to allow the client to initially connect based on capabilities, and I'm not sure we have all the logic to do that yet. So now it is:

```
$ mysql -e "create user u1;set global init_connect='fail';"; mysql -uu1
ERROR 1184 (08S01): Aborted connection 5 to db: 'unconnected' user: 'u1' host: '127.0.0.1' (init_connect command failed)
```